### PR TITLE
mainnet num logic change

### PR DESCRIPTION
**Pull request by GalaxySciTech**
mainnet num logic change

### CODE REVIEW

File: mainnet_contract/contracts/Subnet.sol [hunk 1]
```diff
@@ -240,7 +241,14 @@ contract Subnet {
 
     function setCommittedStatus(bytes32 start_block) internal {
         while ((header_tree[start_block].mix & 1) != 1) {
-            header_tree[start_block].mix |= 1;
+            //change mainnet_num value 0 to block.number
+            header_tree[start_block].mix =
```
Comment (span14) [inline mainnet_contract/contracts/Subnet.sol]: I think this updating method can be simplified as:
`header_tree[start_block.mix] |= block.number << 1`

Comment (GalaxySciTech) [reply mainnet_contract/contracts/Subnet.sol]: Good idea , let me try

Comment (GalaxySciTech) [reply mainnet_contract/contracts/Subnet.sol]: It doesn't seem to work, so finalized can't change from 0 to 1

Comment (GalaxySciTech) [reply mainnet_contract/contracts/Subnet.sol]:
```
            header_tree[start_block].mix |= 1;
            //change mainnet_num value 0 to block.number
            header_tree[start_block].mix |= block.number << 1;
```
            
  is work

**Review by span14 — APPROVED** [review]

**Pull request merged**